batch/test/test_batch.py

```diff
@@ -1583,13 +1576,9 @@ def test_update_with_always_run(client: BatchClient):
     j2 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], always_run=True, parents=[j1])
     b = bb.submit()
 
-    delay = 0.1
-    while True:
-        if j1.is_complete():
-            assert False, str(j1.status(), b.debug_info())
-        if j1.is_running():
-            break
-        delay = sync_sleep_and_backoff(delay)
+    wait_status = j1._wait_for_states('Running')
+    if wait_status.prev_completed:
```

Comment:
Can a job be previously complete but not currently complete?

Reply:
No. This is to distinguish between whether the wait returned because the job was Running or because it had already completed (avoid infinite loops). I don't love this interface, but if you think it's an improvement, then let's go with it. Maybe `prev_completed` is a bad name.

Reply:
Ah ok I see. Instead of adding the extra `prev_completed` field then, can we look at the status dict to see if the state is complete? My initial comment was motivated more by deduplicating the loop-delay logic.